File: app/src/main/java/com/jann_luellmann/thekenapp/dialog/DatePickerFragment.java
package com.jann_luellmann.thekenapp.dialog;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.widget.DatePicker;
import android.widget.EditText;

import java.util.Calendar;
import java.util.Locale;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import lombok.Getter;
import lombok.Setter;

public class DatePickerFragment extends DialogFragment implements DatePickerDialog.OnDateSetListener {

    private EditText dateEditText;

    public DatePickerFragment(EditText dateEditText) {
        this.dateEditText = dateEditText;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        final Calendar c = Calendar.getInstance();
        int year = c.get(Calendar.YEAR);
        int month = c.get(Calendar.MONTH);
        int day = c.get(Calendar.DAY_OF_MONTH);

        return new DatePickerDialog(getActivity(), this, year, month, day);
    }

    @Override
    public void onDateSet(DatePicker datePicker, int year, int month, int day) {
        if(dateEditText != null)
            dateEditText.setText(String.format(Locale.GERMANY, "%d.%d.%d", day, month + 1, year));
    }
}

File: app/src/main/java/com/jann_luellmann/thekenapp/dialog/DatePickerFragment.kt
package com.jann_luellmann.thekenapp.dialog

import android.app.DatePickerDialog
import android.app.Dialog
import android.os.Bundle
import android.widget.DatePicker
import android.widget.EditText
import androidx.fragment.app.DialogFragment
import java.util.*

class DatePickerFragment(dateEditText: EditText?) : DialogFragment(), DatePickerDialog.OnDateSetListener {

    private val dateEditText: EditText? = dateEditText

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val c = Calendar.getInstance()
        val year = c[Calendar.YEAR]
        val month = c[Calendar.MONTH]
        val day = c[Calendar.DAY_OF_MONTH]
        return DatePickerDialog(requireContext(), this, year, month, day)
    }

    override fun onDateSet(datePicker: DatePicker, year: Int, month: Int, day: Int) {
        dateEditText?.setText(
            String.format(
                Locale.GERMANY,
                "%d.%d.%d",
                day,
                month + 1,
                year
            )
        )
    }

}